qiskit_nature/__init__.py

```diff
@@ -49,11 +49,18 @@
 
 """
 
+from typing import Dict, List, Optional, TypeVar, Union
 from .version import __version__
 from .exceptions import QiskitNatureError, UnsupportMethodError
 
+# pylint: disable=invalid-name
+T = TypeVar("T")
+ListOrDictType = Union[List[Optional[T]], Dict[Union[int, str], T]]
```

Comment:
Do you want to do the dict so its different than the minimum eigensolver  - i.e it defines keys as just 'str'? The reason was  to ensure it can be serialized. So if you define int here there arguably be an impedance mismatch when going down the stack whereby perhaps int will not behave as expected - I'm thinking here more around the runtime and remoting these.

Reply:
Yeh you're right. Duplicating this is probably not a good way to go. Unfortunately, Qiskit Terra does not have one ground truth. Instead they have two definitions of `ListOrDict` in `qiskit.algorithms.minimum_eigen_solvers` and in `qiskit.algorithms.eigen_solvers`. While they are currently identical, this is rather error prone and could probably be improved.